src/marqo/api/route.py

```diff
@@ -0,0 +1,25 @@
+from fastapi import Request
+from fastapi.routing import APIRoute
+
+from marqo.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class MarqoCustomRoute(APIRoute):
+    """This is a custom route that logs the error and raises it.
+
+    The log will include the stack trace of the error for debugging purposes.
+    The raised error will be handled by the exception handlers. We DO NOT handle the error here.
+    """
+    def get_route_handler(self):
+        original_route_handler = super().get_route_handler()
+
+        async def marqo_custom_route_handler(request: Request):
```

Comment:
Be very careful about whether this needs to be async. If in doubt just do a normal function.

Reply:
Hi @pandu-k , thanks for being extra care here. Just check the example in FastAPI documentation and this function has to be `async` as the `FastAPI` framework requires the route to be `async`.